File: Data Structures & Algorithms/sliding-window-maximum/submission-5.kt
class Solution {
    fun maxSlidingWindow(nums: IntArray, k: Int): IntArray {
        if (nums.count() == 1) {
            return intArrayOf(nums[0])
        }
        var dequeue = LinkedList<Int>()
        var left = 0
        var right = 0
        var output = mutableListOf<Int>()
        while (right < nums.count()) {
            var current = nums[right]
            while (dequeue.count() > 0 && current > dequeue.last()) {
                dequeue.removeLast()
            }
            dequeue.add(current)
            if (right - left + 1 == k) {
                output.add(dequeue.first())
                if (nums[left] == dequeue.first()) {
                    dequeue.removeFirst()
                }
                left += 1
            }
            right += 1
        }
        return output.toIntArray()
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-5.swift
class Solution {
    func longestConsecutive(_ nums: [Int]) -> Int {
        if nums.isEmpty { return 0 }
        var hash = Set<Int>()
        for num in nums {
            hash.insert(num)
        }
        var i = 0
        var output = 1
        while i < nums.count {
            var current = nums[i]
            if hash.contains(current - 1) {
                i += 1
                continue
            }
            var length = 1
            while hash.contains(current + 1) {
                current = current + 1
                length += 1
                output = max(output, length)
            }
            i += 1
        }
        return output

    }
}
